src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleasePermissionServiceTests.cs

```diff
@@ -527,26 +531,16 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
         [Fact]
         public async Task ListPublicationContributors()
         {
-            var publication = new Publication();
-            var release1 = new Release
-            {
-                Publication = publication,
-                ReleaseName = "2000",
-                TimePeriodCoverage = TimeIdentifier.AcademicYear,
-            };
-            var release2Original = new Release
-            {
-                Publication = publication,
-                ReleaseName = "2001",
-                TimePeriodCoverage = TimeIdentifier.AcademicYear,
-            };
-            var release2Amendment = new Release
-            {
-                Publication = publication,
-                ReleaseName = "2001",
-                TimePeriodCoverage = TimeIdentifier.AcademicYear,
-                PreviousVersion = release2Original,
-            };
+            ReleaseParent releaseParent1 = _dataFixture
+                .DefaultReleaseParent(publishedVersions: 1);
```

Comment:
Just generally thinking about `DefaultReleaseParent`, I think with the arguments it currently takes, while it provides extreme convenience, could be potentially problematic due to the control you're giving up over the test data creation.

I imagine in the future, someone will want to create a parent with several releases that cannot be created with this method. And at that point they either alter the `releaseParent` after creating it via `_dataFixture`, or complicating the `_dataFixture` chain or maybe even having to override `DefaultReleaseParent` or make some other change such that they can customise the releases to get the test data they want.

Maybe that's fine. But I think it is something that needs to be managed carefully as I think this could turn messy quickly. I can imagine problems like lots of different ways of creating objects depending on the amount of customisation you require, or it being hard to follow what a particular `_dataFixture` chain is actually creating.

All that said, I could be overthinking this.

Reply:
> Just generally thinking about DefaultReleaseParent, I think with the arguments it currently takes, while it provides extreme convenience, could be potentially problematic due to the control you're giving up over the test data creation.

I agree.

In some places that we still want to use test data generators for more complicated scenarios it might be better to split up the generation with declarations of different types into separate variables to make the composition a bit clearer, plus it can help avoid long expressions dereferencing particular objects e.g. `releases[0]` over `theme.topics[0].publications[0].releases[0]`.

The use of the new `DefaultReleaseParent(int publishedVersions, bool draftVersion = false, int? year = null)` isn't mandatory either. I just used it in pretty much every change on this PR for convenience.

It's still possible for example to do something like this and keep more control on what extra properties are set:

```csharp
            var releases = _dataFixture
                .DefaultRelease()
                .Generate(1);

            ReleaseParent releaseParent = _dataFixture
                .DefaultReleaseParent()
                .WithReleases(releases);
```

or

```csharp
            Release release = _dataFixture
                .DefaultRelease()
                .WithReleaseParent(_dataFixture
                    .DefaultReleaseParent());
```

> I imagine in the future, someone will want to create a parent with several releases that cannot be created with this method. And at that point they either alter the releaseParent after creating it via _dataFixture, or complicating the _dataFixture chain or maybe even having to override DefaultReleaseParent or make some other change such that they can customise the releases to get the test data they want.

I agree, I think they will. I haven't spent too much time thinking about what the signature of it should be and what the possible use cases of it are. I did seem to be able to retrofit it into pretty much all the unit tests I wanted to update as part of this PR though without too much complication.

>  I can imagine problems like lots of different ways of creating objects depending on the amount of customisation you require, or it being hard to follow what a particular _dataFixture chain is actually creating.

I agree, if there's a lot of customisation then it's probably best to declare separate variables with customisation first and compose a hierarchy from those if necessary.

We can use of `ToTuple2` and `ToTuple3` to dereference created objects with explicit variable naming to help avoid having to understand the configuration, e.g.

```csharp
            var (publication, unpublishedPublication) = _dataFixture
                .DefaultPublication()
                ...
                .Generate(2)
                .ToTuple2();
```